### ISSUE TITLE
Import fails due to missing pkg_resources

### ISSUE THREAD

**Issue by BastianZim** [PennyLaneAI/pennylane-honeywell#54]
Hi,

I am building the package for conda-forge and here the import fails with the following error message:

```shell

import: 'pennylane_honeywell'
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/pennylane-honeywell_1698860547974/test_tmp/run_test.py", line 2, in <module>
    import pennylane_honeywell
  File "/home/conda/feedstock_root/build_artifacts/pennylane-honeywell_1698860547974/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/pennylane_honeywell/__init__.py", line 19, in <module>
    from .device import HQSDevice
  File "/home/conda/feedstock_root/build_artifacts/pennylane-honeywell_1698860547974/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/pennylane_honeywell/device.py", line 30, in <module>
    import pennylane as qml
  File "/home/conda/feedstock_root/build_artifacts/pennylane-honeywell_1698860547974/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/pennylane/__init__.py", line 21, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

What could be the reason for this?

Logs: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=815629&view=logs&jobId=656edd35-690f-5c53-9ba3-09c10d0bea97&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3
PR: https://github.com/conda-forge/pennylane-honeywell-feedstock/pull/2
Deps: https://github.com/regro-cf-autotick-bot/pennylane-honeywell-feedstock/blob/0.33.0_h1476dc/recipe/meta.yaml

**Comment by mlxd** [comment]
Hey @BastianZim thanks for the report. I assume this is a Python 3.12 build?
For 3.12, the CPython maintainers have removed `setuptools` from being default installed (see https://github.com/python/cpython/issues/95299), and so our previous assumption of this always being available is no longer the case on 3.12

We have a PR opened to add this into the master branch (see https://github.com/PennyLaneAI/pennylane/pull/4771 which was opened not long ago), which will allow us to enable 3.12 support for the next release.

For now,  we do not officially support Python 3.12, and so for a temporary working solution you can ensure `setuptools` is pre-installed in the environment you are working in. This should bring in the `pkg_resources` module, which is required for device loading. Feel free to let us know if this helps.

**Comment by BastianZim** [comment]
That was it – thank you!
